### ISSUE TITLE
gnome-shell 3.26.2 crashes when notification pops up

### ISSUE THREAD

**Issue by tomasz-oponowicz** [codito/gnome-pomodoro#320]
From time to time gnome-shell 3.26.2 crashes and restarts when desktop or screen notification pops up.

Logs:
```
Process 1102 (gnome-shell) of user 1000 dumped core.

Stack trace of thread 1102:
#0  0x00007f698038dd3f g_type_check_instance_is_fundamentally_a (libgobject-2.0.so.0)
#1  0x00007f698036f870 g_value_set_object (libgobject-2.0.so.0)
#2  0x00007f6980373945 g_object_get_property (libgobject-2.0.so.0)
#3  0x00007f697f03d2fe n/a (libgjs.so.0)
#4  0x00007f6978465c5b n/a (libmozjs-52.so.0)
#5  0x00007f697848dfcb n/a (libmozjs-52.so.0)
#6  0x00007f697818bd05 n/a (libmozjs-52.so.0)
#7  0x000039bda7eb1f06 n/a (n/a)
#8  0x00005583d7d5f0e8 n/a (n/a)
#9  0x000039bda804cfd5 n/a (n/a)
```

In order to make sure I disabled gnome-pomodoro extension for some time and the issue didn't occur.

**Comment by tomasz-oponowicz** [comment]
The problem occurs in 13.2, 13.3 and master (e472a4dca4e1c779b36a0d9d358bfeadfb7e17bc).

**Comment by zkanda** [comment]
I'm experiencing the same issue.

**Comment by porjo** [comment]
FYI, this Redhat Bugzilla report has Pomodoro as the prime suspect in a lot of reported crashes: https://bugzilla.redhat.com/show_bug.cgi?id=1494586

**Comment by berenddeschouwer** [comment]
Possible duplicate of issue #314

**Comment by 3v1n0** [comment]
Just disconnect from these signals https://github.com/codito/gnome-pomodoro/blob/0.13.3/plugins/gnome/extension/dialogs.js#L105 when this.actor is destroyed, this should be enough to fix it.

**Comment by zkanda** [comment]
@3v1n0 maybe you can send a pull request?

**Comment by berenddeschouwer** [comment]
Wrote a pull request (which ended up as issue #321... how do I fix that?)

It's against the gnome-3.26 branch.  Master does not merge cleanly.

It implements what 3v1n0 suggested.  It seems to WorkForMe(tm).

I don't know how to do a full test, so I suggest further testing.

**Comment by kamilprusko** [comment]
Indeed, seems this solves the issue.

Patch is included in [0.13.4 release](https://github.com/codito/gnome-pomodoro/releases/tag/0.13.4), .debs are [here](https://download.opensuse.org/repositories/home:/kamilprusko/).

Great thanks to everyone for the help!
